src/main/cpp/util/md5.cc

```diff
@@ -209,10 +210,11 @@ void Md5Digest::Transform(
   // FF, GG, HH, and II transformations for rounds 1, 2, 3, and 4.
   // Rotation is separate from addition to prevent recomputation.
 #define FF(a, b, c, d, s, ac) { \
-      (a) += F((b), (c), (d)) + ((*x_pos++ = *cur_word++)) + \
+      (a) += F((b), (c), (d)) + ((*x_pos++ = htole32(*cur_word))) + \
           static_cast<uint32_t>(ac); \
       (a) = ROTATE_LEFT((a), (s)); \
       (a) += (b); \
+      cur_word++; \
```

Comment:
This is to avoid the double evaluation of the postfix increment operator, right?

Reply:
Yes. Just to make sure that it is side effect free.